addons/account/views/account_move_views.xml

```diff
@@ -1346,6 +1346,10 @@
                                                context="{'default_partner_id': bank_partner_id, 'display_account_trust': True}"
                                                domain="[('partner_id.ref_company_ids', 'parent_of', company_id)]"
                                                readonly="state != 'draft'"/>
+                                        <field name="payment_reference"
+                                            invisible="move_type not in ('out_invoice', 'out_refund')"
+                                            readonly="inalterable_hash != False"
+                                            placeholder="Use Bill Reference"/>
```

Comment:
The placeholder here doesn't make sense. Replace by "Standard communication" instead, which will implicitely refer to the configuration made at the Journal level.

Otherwise, LGTM !